ring/util_test.go

```diff
@@ -182,44 +194,40 @@ func TestWaitRingStabilityShouldReturnOnceMinStabilityHasBeenReached(t *testing.
 	assert.LessOrEqual(t, elapsedTime.Milliseconds(), (minStability + addInstanceAfter + 3*time.Second).Milliseconds())
 }
 
-func TestWaitRingStabilityShouldReturnErrorIfMaxWaitingIsReached(t *testing.T) {
+func TestWaitRingTokensStability_ShouldReturnOnceMinStabilityOfInstancesHasBeenReached(t *testing.T) {
 	t.Parallel()
 
 	const (
-		minStability = 2 * time.Second
-		maxWaiting   = 7 * time.Second
+		minStability     = 3 * time.Second
+		addInstanceAfter = 2 * time.Second
+		maxWaiting       = 15 * time.Second
 	)
 
-	// Init the ring.
-	ringDesc := &Desc{Ingesters: map[string]InstanceDesc{
-		"instance-1": {Addr: "instance-1", State: ACTIVE, Timestamp: time.Now().Unix()},
-		"instance-2": {Addr: "instance-2", State: PENDING, Timestamp: time.Now().Unix()},
-		"instance-3": {Addr: "instance-3", State: JOINING, Timestamp: time.Now().Unix()},
-		"instance-4": {Addr: "instance-4", State: LEAVING, Timestamp: time.Now().Unix()},
-		"instance-5": {Addr: "instance-5", State: ACTIVE, Timestamp: time.Now().Unix()},
-	}}
+	ring := createStartingRing()
 
-	ring := &Ring{
-		cfg:                 Config{HeartbeatTimeout: time.Minute},
-		ringDesc:            ringDesc,
-		ringTokens:          ringDesc.GetTokens(),
-		ringTokensByZone:    ringDesc.getTokensByZone(),
-		ringInstanceByToken: ringDesc.getTokensInfo(),
-		ringZones:           getZones(ringDesc.getTokensByZone()),
-		strategy:            NewDefaultReplicationStrategy(),
-	}
+	// Add 1 new instance after some time.
+	go addInstanceAfterSomeTime(ring, addInstanceAfter)
```

Comment:
This function already starts extra goroutine, so this `go` keyword is currently extra.